File: frontend/src/types/caseBrainSearch.ts
export interface CaseBrainSearchEntry {
  brain_entry_id: number;
  matter_key: string;
  email_id: string | null;
  occurred_at: string;
  logged_at: string;
  source_type: string;
  source_reference: string | null;
  source_actor: string | null;
  update_summary: string;
  logged_by: string | null;
}

export interface CaseBrainSearchResponse {
  total: number;
  limit: number;
  offset: number;
  entries: CaseBrainSearchEntry[];
}

File: backend/app/schemas/case_brain_search.py
"""
Pydantic schemas for the firm-wide Case Brain search endpoint.

GET /api/case-brain
"""

from __future__ import annotations

from typing import List, Optional

from pydantic import BaseModel


class CaseBrainSearchEntry(BaseModel):
    """A single Case Brain entry in the firm-wide search response."""

    brain_entry_id: int
    matter_key: str
    email_id: Optional[str] = None
    occurred_at: str
    logged_at: str
    source_type: str
    source_reference: Optional[str] = None
    source_actor: Optional[str] = None
    update_summary: str
    logged_by: Optional[str] = None


class CaseBrainSearchResponse(BaseModel):
    """Paginated firm-wide Case Brain timeline."""

    total: int
    limit: int
    offset: int
    entries: List[CaseBrainSearchEntry]
